map/framework.cpp

```diff
@@ -517,10 +517,15 @@ bool Framework::OnCountryFileDelete(storage::TCountryId const & countryId, stora
 
 void Framework::OnMapDeregistered(platform::LocalCountryFile const & localFile)
 {
-  GetPlatform().RunOnGuiThread([this, localFile]
+  auto action = [this, localFile]
   {
     m_storage.DeleteCustomCountryVersion(localFile);
-  });
+  };
+  
+  if (m_storage.GetThreadChecker().CalledOnOriginalThread())
```

Comment:
Выглядит как непонятный костыль. Хотя бы задокументируйте, почему это сделано и для чего.

> On 25.3.2016, at 20.47, ygorshenin <notifications@github.com> wrote:
> 
> In map/framework.cpp:
> 
> >    {
> >      m_storage.DeleteCustomCountryVersion(localFile);
> > -  });
> > +  };
> > +  
> > +  if (m_storage.GetThreadChecker().CalledOnOriginalThread())
> Нет, так как только Framework знает на котором потоке был создан storage. storage не знает на каком потоке он был создан, и создан он быть может на любом потоке, и в тестах он как раз содается на любых потоках. А вот Framework знает что создается на GUI потоке, и, как следствие, в случае несоответствия обладает всесми возможностями для того чтобы вызвать правильный метод storage на правильном потоке.
> 
> —
> You are receiving this because you commented.
> Reply to this email directly or view it on GitHub
>